localflavor/by/forms.py

```diff
@@ -0,0 +1,102 @@
+from django import forms
+from django.utils.translation import ugettext_lazy as _
+
+from .by_regions import BY_REGIONS_CHOICES
+
+
+class BaseKwargsUpdatedField:
+    """
+    Abstract base class made to conform the DRY principle.
+
+    The initial_options is overrided by the dict in the subclasses and then
+    automatically passed to kwargs in the FormField.__init__() method.
+    """
+
+    initial_options = None
+
+    def __init__(self, *args, **kwargs):
+        kwargs.update(self.initial_options)
+        super().__init__(*args, **kwargs)
+
+
+class UpperValueMixin:
+    """
+    Mixin made to conform the DRY principle.
+
+    Overrides the to_python methods to return the value.upper()
+    """
+
+    def to_python(self, value):
+        value = super().to_python(value)
+        if value is None:
+            return value
+        else:
+            return value.upper()
+
+
+class BYRegionField(BaseKwargsUpdatedField, forms.TypedChoiceField):
```

Comment:
Instead of making `ChoiceField` for the regions, I think it's better to make a `forms.Select` widget that can be used in forms. The `BY_REGIONS_CHOICES` can be set directly on the model instead of using this custom form field.

Here's an example of a `forms.Select` widget:
https://github.com/django/django-localflavor/blob/456dd7e44537a3170c5d672dd51481885b0ff5e8/localflavor/md/forms.py#L35-L43